usdigitalresponse/usdr-gost
# fix: #1676 - Cache prior reporting periods

This is a first commit of a PR to cache prior reporting periods. This splits the report into two pieces such that the generate function should only need to call the most recent period. This should help with our reporting bottlenecks and with our performance issues.

### Ticket #1676
## Description

## Screenshots / Demo Video
No screenshots at this time as it is only backend functionality.

## Testing
This can be tested by running locally, opening the prior reporting period, closing the reporting period, then generating the report. To compare, turn cache off locally and rerun the report. Then compare to make sure they are the same.

### Automated and Unit Tests

### Manual tests for Reviewer
To test, run a report locally. With logging on, you will be able to see `cache hit` or `cache miss` in the logs.
The reports should look identical in both cases. They should also look identical to a report without this change.

## Checklist
